jgrapht-core/src/main/java/org/jgrapht/alg/flow/PushRelabelMaximumFlow.java

```diff
@@ -0,0 +1,388 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* -----------------
+ * PushRelabelMaximumFlow.java
+ * -----------------
+ * (C) Copyright 2015-2015, by Alexey Kudinkin and Contributors.
+ *
+ * Original Author:  Alexey Kudinkin
+ * Contributor(s):
+ *
+ * $Id$
+ *
+ * Changes
+ * -------
+ */
+package org.jgrapht.alg.flow;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.alg.util.Extension.ExtensionFactory;
+import org.jgrapht.alg.util.Pair;
+
+import java.util.*;
+
+/**
+ * <p><a href="https://en.wikipedia.org/wiki/Push%E2%80%93relabel_maximum_flow_algorithm">
+ * Push-relabel maximum flow</a> algorithm designed by Andrew V. Goldberg and
+ * Robert Tarjan. Current implementation complexity upper-bound is O(V^3). For more
+ * details see: <i>"A new approach to the maximum flow problem"</i> by Andrew V. Goldberg
+ * and Robert Tarjan <i>STOC '86: Proceedings of the eighteenth annual ACM symposium on
+ * Theory of computing</i></p>
+ *
+ * @author Alexey Kudinkin
+ */
+public class PushRelabelMaximumFlow<V, E> extends MaximumFlowAlgorithmBase<V,E> {
+
+    private static final boolean LABEL_PRUNE_ENABLED = true;
```

Comment:
Either this should be removed together with corresponding conditionals, or else it should be made into an actual parameter (with test coverage for true/false settings).